'set' as TSV column name causes errors

I have a dataset that has a `scans.tsv` file within each session. The contents of this file are something like:
```
filename	task	set	order
func/sub-M2_ses-20100911_task-dkl_run-01_bold.nii.gz	dkl	1	1
func/sub-M2_ses-20100911_task-dkl_run-02_bold.nii.gz	dkl	1	1
func/sub-M2_ses-20100911_task-dkl_run-03_bold.nii.gz	dkl	1	2
func/sub-M2_ses-20100911_task-dkl_run-04_bold.nii.gz	dkl	1	3
.
.
.
```
Where `set` and `order` refer to different arrangements of stimulus presentation blocks. When I run the validator (`deno run -A jsr:@bids/validator bids_dataset`). I get the following errors for this file:
```
[ERROR] SCANS_FILENAME_NOT_MATCH_DATASET Filenames in scans.tsv file do not match what is present in the BIDS dataset.
[ERROR] TSV_COLUMN_MISSING A required column is missing
		filename
```
I know for a fact that the scans filenames do match the contents of the `func` directory, and you can see that the TSV has the `filename` column.

@ericearl suggested that I change the `set` column to something else, so I made the following change:
```
filename	task	condition	order
func/sub-M2_ses-20100911_task-dkl_run-01_bold.nii.gz	dkl	1	1
func/sub-M2_ses-20100911_task-dkl_run-02_bold.nii.gz	dkl	1	1
func/sub-M2_ses-20100911_task-dkl_run-03_bold.nii.gz	dkl	1	2
func/sub-M2_ses-20100911_task-dkl_run-04_bold.nii.gz	dkl	1	3
.
.
.
```
and this does not give any errors. It appears as though the string `set` is privileged within the validator and should either be fixed or documented.

--- a/src/types/columns.test.ts
+++ b/src/types/columns.test.ts
@@ -37,6 +37,13 @@ Deno.test('ColumnsMap', async (t) => {
     // @ts-expect-error ibid
     assertEquals(columns.size, ['0'])
   })
+  await t.step('set columns are permissible', () => {
+    const columns = new ColumnsMap()
+    // @ts-expect-error ts thinks size is protected property
+    columns['set'] = ['0']
+    // @ts-expect-error ibid
+    assertEquals(columns.set, ['0'])
+  })
   await t.step('missing columns are undefined', () => {
     const columns = new ColumnsMap()
     columns['a'] = ['0']

--- a/src/types/columns.ts
+++ b/src/types/columns.ts
@@ -15,8 +15,14 @@ const columnMapAccessorProxy = {
     prop: symbol | string,
     receiver: ColumnsMap,
   ) {
-    // Map.size exists, so we need to shadow it with the column contents
-    if (prop === 'size') return target.get('size')
+    // Map instance methods/properties that could plasubily be column names:
+    if (
+      ['clear', 'delete', 'keys', 'set', 'values', 'size'].includes(
+        prop as string,
+      )
+    ) {
+      return target.get(prop as string)
+    }
     const value = Reflect.get(target, prop, receiver)
     if (typeof value === 'function') return value.bind(target)
     if (prop === Symbol.iterator) return target[Symbol.iterator].bind(target)

